tests/component/test_task.py

```diff
@@ -26,3 +40,61 @@ def test___tasks_with_different_names___hash___not_equal(generate_task):
     task2 = generate_task("MyTask2")
 
     assert hash(task1) != hash(task2)
+
+
+@pytest.mark.device_name("bridgeTester")
+@pytest.mark.parametrize(
+    "shunt_resistor_value, shunt_resistor_location, shunt_resistor_select, shunt_resistor_source, bridge_resistance, skip_unsupported_channels",
+    [
+        (100000, 12465, 0, 0, 0.2, True),
+        (100000, 12465, 0, 0, 0.2, False),
+        (100000, 12467, 0, 0, 0.2, True),
+        (100000, 12467, 0, 0, 0.2, False),
+    ],
+)
+def test___perform_bridge_shunt_cal___no_errors(
```

Comment:
Are these strain cal functions guaranteed to succeed on simulated devices? This test had one failure that wasn't related to gRPC.

https://github.com/ni/nidaqmx-python/actions/runs/7444178717/job/20285452399?pr=469#step:4:272

```
tests\component\test_task.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
generated\nidaqmx\task.py:476: in perform_bridge_shunt_cal
    self._interpreter.perform_bridge_shunt_cal_ex(
generated\nidaqmx\_library_interpreter.py:3898: in perform_bridge_shunt_cal_ex
    self.check_for_error(error_code)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <nidaqmx._library_interpreter.LibraryInterpreter object at 0x03DF3F08>
error_code = -201493, samps_per_chan_written = None, samps_per_chan_read = None

    def check_for_error(self, error_code, samps_per_chan_written=None, samps_per_chan_read=None):
        if not error_code:
            return
    
        if error_code < 0:
            extended_error_info = self.get_extended_error_info()
    
            if samps_per_chan_read is not None:
                raise DaqReadError(extended_error_info, error_code, samps_per_chan_read)
            elif samps_per_chan_written is not None:
                raise DaqWriteError(extended_error_info, error_code, samps_per_chan_written)
            else:
>               raise DaqError(extended_error_info, error_code)
E               nidaqmx.errors.DaqError: Shunt calibration failed. The calculated gain adjust is out of range. Ensure that the shunt calibration terminals are connected properly and that the shunt resistance and shunt element location settings match how the hardware is wired.
E               Property: DAQmx_AI_Bridge_ShuntCal_ShuntCalAResistance
E               Corresponding Value:  100.0e3
E               Simulated Value with Shunt: -1.249999e-6
E               Measured Value with Shunt: -2.717376e-6
E               Property: DAQmx_AI_Bridge_ShuntCal_GainAdjust
E               Requested Value:  460.002117e-3
E               Maximum Value:  1.500000
E               Minimum Value:  500.0e-3
E               Channel Name: bridgeTester/ai0
E               
E               Task Name: _unnamedTask<18>
E               
E               Status Code: -201493

generated\nidaqmx\_library_interpreter.py:6080: DaqError
```

Reply:
No, ShuntCal will fail with the error you pasted if you don't actually have the hardware configuration wired up correctly. I guess it could randomly pass if the simulated data works out. Its always a sine wave (in ADC codes I think?), which goes haywire with Strain, I believe.  I guess maybe you could randomly get an average close to expected and it would succeed, but I highly doubt it. I think making this test expect that error is probably fine.

Reply:
I think most applications perform offset nulling before shunt cal. That may also make a difference. But yeah, this test ought to tolerate this error.

Reply:
If the test always expected an error, the answer would be to use `with pytest.raises(...):`

Since we want to tolerate some errors, we probably need to do
```
try:
   task.whatever()
except DaqError as e:
    if e.code != whatever:
        raise
```